### PR TITLE
[GH-12291] UI Automation: Write an automated test using Cypress for "Markdown quotation paragraphs"

### PR DESCRIPTION
**Pull request by larkox**
#### Summary

UI Automation: Write an automated test using Cypress for "Markdown quotation paragraphs"
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12291
#### Related Pull Requests

None

#### Screenshots

None

### CODE REVIEW

File: e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                    // * There is only one blockquote, and therefore only one Quote icon
+                    cy.get(`#postMessageText_${postId} > blockquote`).should('have.length', 1);
+
+                    // * There are three distinct paragraphs, and therefore the three of them are separated by a space
+                    cy.get(`#postMessageText_${postId} > blockquote > p`).should('have.length', 3);
+                    cy.get(`#postMessageText_${postId} > blockquote > p`).each((el, i) => {
+                        // * Each paragraph contains the content we put on the message
+                        expect(messageParts[i]).equals(el.html());
+
+                        // # We save the alignment of the firts paragraph
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js]:
```suggestion
                        // # We save the alignment of the first paragraph
```

Comment (larkox) [reply e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js]: This and below done on 6f5615c . Thanks!

File: e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        // # Post message to use
+        cy.postMessage(message);
+
+        var firstPartLeft;
+        cy.getLastPostId().then((postId) => {
+            // * There is only one blockquote, and therefore only one Quote icon
+            cy.get(`#postMessageText_${postId} > blockquote`).should('have.length', 1);
+
+            // * There are three distinct paragraphs, and therefore the three of them are separated by a space
+            cy.get(`#postMessageText_${postId} > blockquote > p`).should('have.length', 3);
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js]: For some reason, this currently failed on me, probably related to recent Cypress upgrade.  Could please check again?  Thanks!

Comment (larkox) [reply e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js]: Apparently, line breaks on 3.4 where considered as line breaks and on 3.6 as hitting on enter. Therefore, I had to change them so I use the {shift}{enter} way.

Also, clear is working fine on 3.6. Therefore, probably was not a problem with the platform but the version of Cypress.

Thanks!

Comment (saturninoabril) [reply e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js]: Thanks for checking!

File: e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        // # Post message to use
+        cy.get('#post_textbox').clear().type('>' + messageParts[0]);
+        cy.get('#post_textbox').type('{shift}{enter}{enter}');
+
+        cy.get('#post_textbox').type('>' + messageParts[1]);
+        cy.get('#post_textbox').type('{shift}{enter}{enter}');
+
+        cy.get('#post_textbox').type('>' + messageParts[2]);
+        cy.get('#post_textbox').type('{enter}');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js]: Last request and I'm all good to merge.  Please change to
```javascript
cy.get('#post_textbox').clear().
    type('>' + messageParts[0]).type('{shift}{enter}{enter}').
    type('>' + messageParts[1]).type('{shift}{enter}{enter}').
    type('>' + messageParts[2]).type('{enter}');
```

File: e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    it('M18703-Markdown quotation paragraphs', () => {
+        const messageParts = ['this is', 'really', 'three quote lines'];
+
+        cy.visit('/ad-1/channels/town-square');
+
+        // # Post message to use
+        cy.get('#post_textbox').clear().type('>' + messageParts[0]).type('{shift}{enter}{enter}');
+        cy.get('#post_textbox').type('>' + messageParts[1]).type('{shift}{enter}{enter}');
+        cy.get('#post_textbox').type('>' + messageParts[2]).type('{enter}');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/markdown_quotation_paragraphs_spec.js]: I'm good with this as is but would be nice to access `cy.get('#post_textbox')` once, and chain others as necessary.

**Pull request merged**